src/mruby-zest/example/TextLine.qml

```diff
@@ -50,28 +50,72 @@ Widget {
         (0...l.length).each do |i|
             l[i] = "?" if l.getbyte(i) > 127
         end
-        vg.text(8,h/2,l)
-        bnd = vg.text_bounds(0,0,l)
-        if(@state)
-            vg.text(8+bnd,h/2,"|")
+
+        @edit ||= EditRegion.new($vg, self.label, w-20, h*0.8)
+        @edit.each_string do |x, y, str, cursor|
+            if(cursor == false)
+                vg.text(x+10, y, str)
+            else
+                if(@state)
+                    vg.text_align NVG::ALIGN_LEFT| NVG::ALIGN_MIDDLE
+                    vg.text(x+10, y, str)
+                end
+            end
         end
+
     }
 
     function onKey(k, mode)
     {
         return if mode != "press"
+        pos = self.label.length
+        pos = @edit.pos if @edit
+        ll = self.label
+
+        cursor_row = @edit.calc_cursor_y
+
         if(k.ord == 8)
-            self.label = self.label[0...-1]
-        elsif k.ord >= 32
```

Comment:
This line disappeared with your change, and it's important. This is how TextLine ignores most of the non-printable characters.

Reply:
Hi @pgervais ,
This line is the same as self.label = ll[0...-1], because of that I removed it and tried to replace my line with the previous line and the result was the same.

Reply:
Just to double check we're talking about the same thing. I was referring to the "elsif k.ord >= 32" line. Is that what you meant?

Reply:
Possibly an off-topic comment, but make sure to use https://github.com/mruby-zest/mruby-zest-build/commit/a5810360f86ef7bc18825ec0f2271eb9797e9ae1 in your branch if you're experiencing problems with creating shorter strings.

Reply:
Hi Phillip,
No, I mean the self.label line, but I used the else instead of this line (elsif k.ord >= 32) but I realized that in the else I include the case where k.ord between 8 and 32, so I think I need to modify it as you say, what do you think?

Reply:
You can implement it the way you prefer. What is most important is to make sure values of k.ord less than 32 are ignored (with the exception of 8)